crates/noirc_evaluator/src/ssa_refactor/acir_gen/mod.rs

```diff
@@ -110,7 +110,10 @@ impl Context {
             _ => unreachable!("ICE: Program must have a singular return"),
         };
 
-        let is_return_unit_type = return_values.len() == 1 && dfg.type_of_value(return_values[0]) == Type::Unit;
+        // Check if the program returns the `Unit/None` type.
+        // This type signifies that the program returns nothing.
+        let is_return_unit_type =
```

Comment:
A thought: We could probably move this `is_return_unit_type` function into the dfg and call `dfg.return_is_unit_type` -- The benefit being that if the ssa ir did decide to make programs have no return values for Unit, we would not need to change this piece of code to something like:

```
let is_return_unit_type = return_values.len() == 0
```

It would just be encapsulated inside of the ssa IR code

Reply:
https://github.com/noir-lang/noir/issues/1438